File: spa/lib/models/custom_content.dart
class CustomContentBlock {
  final int id;
  final String title;
  final String? subtitle;
  final String? description;
  final String? imageUrl;
  final String? actionUrl;
  final String? actionText;
  final String blockType;
  final int orderIndex;
  final bool isActive;
  final String? backgroundColor;
  final String? textColor;
  final String? gradientStart;
  final String? gradientEnd;

  CustomContentBlock({
    required this.id,
    required this.title,
    this.subtitle,
    this.description,
    this.imageUrl,
    this.actionUrl,
    this.actionText,
    required this.blockType,
    required this.orderIndex,
    required this.isActive,
    this.backgroundColor,
    this.textColor,
    this.gradientStart,
    this.gradientEnd,
  });

  factory CustomContentBlock.fromJson(Map<String, dynamic> json) {
    return CustomContentBlock(
      id: json['id'] as int,
      title: json['title'] as String,
      subtitle: json['subtitle'] as String?,
      description: json['description'] as String?,
      imageUrl: json['image_url'] as String?,
      actionUrl: json['action_url'] as String?,
      actionText: json['action_text'] as String?,
      blockType: json['block_type'] as String,
      orderIndex: json['order_index'] as int,
      isActive: json['is_active'] as bool,
      backgroundColor: json['background_color'] as String?,
      textColor: json['text_color'] as String?,
      gradientStart: json['gradient_start'] as String?,
      gradientEnd: json['gradient_end'] as String?,
    );
  }
}

File: backend/app/models/custom_content.py
"""
Модель для кастомных блоков контента на главном экране
"""
from sqlalchemy import Column, String, Text, Integer, Boolean, Enum as SQLEnum
from app.models.base import BaseModel
import enum


class ContentBlockType(str, enum.Enum):
    """Типы блоков контента.

    Наследование от str нужно SQLAlchemy 2: для обычного enum.Enum в БД по умолчанию
    пишутся имена (SPA_TRAVEL), а в PostgreSQL у нас значения ('spa_travel', …).
    """
    SPA_TRAVEL = "spa_travel"  # Spa-путешествия
    PROMOTION = "promotion"  # Акция
    BANNER = "banner"  # Баннер
    CUSTOM = "custom"  # Кастомный блок


def _enum_values(enum_cls):
    return [member.value for member in enum_cls]


class CustomContentBlock(BaseModel):
    """Кастомный блок контента для главного экрана"""
    __tablename__ = "custom_content_blocks"
    
    title = Column(String(200), nullable=False)  # Заголовок блока
    subtitle = Column(String(300), nullable=True)  # Подзаголовок
    description = Column(Text, nullable=True)  # Описание (может быть HTML)
    image_url = Column(String(500), nullable=True)  # URL изображения
    action_url = Column(String(500), nullable=True)  # URL для действия (ссылка)
    action_text = Column(String(100), nullable=True)  # Текст кнопки действия
    block_type = Column(
        SQLEnum(
            ContentBlockType,
            name="contentblocktype",
            values_callable=_enum_values,
            native_enum=False,
        ),
        default=ContentBlockType.CUSTOM,
        nullable=False,
    )
    order_index = Column(Integer, default=0, nullable=False)  # Порядок отображения
    is_active = Column(Boolean, default=True, nullable=False)  # Активен ли блок
    background_color = Column(String(7), nullable=True)  # HEX цвет фона
    text_color = Column(String(7), nullable=True)  # HEX цвет текста
    gradient_start = Column(String(7), nullable=True)  # HEX цвет начала градиента
    gradient_end = Column(String(7), nullable=True)  # HEX цвет конца градиента
    
    def __repr__(self):
        return f"<CustomContentBlock {self.title}>"
